dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueClients.java

```diff
@@ -128,6 +128,26 @@ public interface StickyChannelFactory2 {
         Channel getStickyChannel();
 
         <T> T sticky(Class<T> clientInterface);
+
+        /**
+         * Creates a new {@link StickyChannelSession} which can be used to create multiple
+         * clients bound to the same session.
+         */
+        StickyChannelSession session();
+    }
+
+    /**
+     * Represents a single session of {@link StickyChannelFactory2} where the underlying channel will not change.
+     * This is useful when multiple client interfaces must be bound to use the same host.
+     */
+    public interface StickyChannelSession {
```

Comment:
Naming: should this interface have a 2 somewhere? Unlike the sticky channels factory, it is meant to be stored for reuse, so there's upside to having a cleaner name. I don't have a strong opinion either way.